src/Classes/iTones/iTones_Playlist.php

```diff
@@ -513,6 +548,78 @@ public static function getPlaylistFromWeights($playlists_to_ignore = [])
         return self::getInstance(CoreUtils::biasedRandom($result));
     }
 
+    /**
+     * Uses weighted playout values to select a random Playlist from a category, returning it.
+     *
+     * Only includes Playlists with a currently running slot, and a Track.
+     *
+     * @param int $categoryId
+     * @param array $playlists_to_ignore one or more playlists to not return
+     * @return iTones_Playlist
+     */
+    public static function getPlaylistOfCategoryFromWeights($categoryId, $playlists_to_ignore = [])
+    {
+        // TODO: this is a straight copy-paste of the above. If we need to do this again,
+        // consider refactoring.
+        self::wakeup();
+
+        $result = self::$db->fetchAll(
+            'SELECT playlists.playlistid AS item, MAX(playlist_availability.weight) AS weight
+                FROM jukebox.playlists, jukebox.playlist_availability, jukebox.playlist_timeslot
+                WHERE playlists.category = $1
```

Comment:
Is this function actually needed? Could we not just put categoryId param with a default of "*" in the regular getPlaylistFromWeights function?

Reply:
That's a bit too close to SQL injection for my liking: we'd have to dynamically construct the SQL if the param is not present - somewhat tame as far as things go, but I'm not wholly convinced of the need. If for some reason we add a third function that does a substantially similar query, then we can look at this again.

Reply:
I mean, like you can do funcname([...], $categoryId == "*"), meaning if it's not provided, it's "*". Then, if the value isn't === "*", you check if it's an integer, before putting it into the SQL query. Because afaict the only difference between this and the other function is the "WHERE playlists.category = $1" bit. You can just do "WHERE playlists.category = *" in this case.

Reply:
Sadly, that's not a thing you can do.

![image](https://user-images.githubusercontent.com/2904440/71352271-2ed06780-2576-11ea-8fcc-345a510391c8.png)

Reply:
aww.